### PR TITLE
feat(api): add pagination capabilities with middleware

### PR DESCRIPTION
**Pull request by GMishx**
Add new middleware to handle pagination.

Handle pagination for `/licenses`, `/obligations` and `/audits` endpoints.

Also update the `PaginationMeta` to contain the `Totalpages`, `Page`, `Limit`, `Previous` and `Next` information.

To merge after #32

### CODE REVIEW

**Review by avinal — COMMENTED** [review]
The rest looks good to me.

File: pkg/models/types.go [hunk 1]
```diff
@@ -87,19 +87,44 @@ type LicenseUpdate struct {
...
 type PaginationMeta struct {
-	ResourceCount int `json:"resource_count" example:"20"`
-	Page          int `json:"page,omitempty" example:"1"`
-	PerPage       int `json:"per_page,omitempty" example:"10"`
+	ResourceCount int    `json:"resource_count" example:"200"`
+	TotalPages    int    `json:"total_pages,omitempty" example:"20"`
+	Page          int    `json:"page,omitempty" example:"10"`
+	Limit         int    `json:"limit,omitempty" example:"10"`
+	Next          string `json:"next,omitempty" example:"/api/v1/licenses?limit=10&page=11"`
+	Previous      string `json:"previous,omitempty" example:"/api/v1/licenses?limit=10&page=9"`
```
[the remark this answers is not in the document]
Comment (GMishx) [reply pkg/models/types.go:95]: No particular reason. Can be helpful for the UI to navigate easily. Also at the edges, where there is no next/previous, will be helpful.

[the remark this answers is not in the document]
Comment (avinal) [reply pkg/models/types.go:95]: Understood.

**Review by avinal — APPROVED** [review]
LGTM

**Pull request merged**